app/src/main/java/com/bignerdranch/android/geoquiz/MainActivity.kt

```diff
@@ -121,22 +122,68 @@ class MainActivity : AppCompatActivity() {
     private fun checkAnswer(userAnswer: Boolean) {
         val correctAnswer = quizViewModel.currentQuestionAnswer
 
+        trueButton.isEnabled = false
+        falseButton.isEnabled = false
+        quizViewModel.questionIsAnswered()
+
         val messageResId = when {
             quizViewModel.isCheater -> R.string.judgment_toast
             userAnswer == correctAnswer -> R.string.correct_toast
             else -> R.string.incorrect_toast
         }
 
-        val toast = Toast.makeText(this, messageResId, Toast.LENGTH_SHORT)
-        toast.setGravity(Gravity.TOP, 0, 300)
-        toast.show()
+        if (userAnswer == correctAnswer) {
+            pointForCorrectAnswer()
+        }
+
+        val toastAnswerStatus = Toast.makeText(
+            this,
+            messageResId,
+            Toast.LENGTH_SHORT
+        )
+        toastAnswerStatus.setGravity(Gravity.TOP, 0, 300)
+        toastAnswerStatus.show()
+
+        val scoreToast = Toast.makeText(
+                this,
+                "Twój wynik to " +
+                        "${(quizViewModel.score / quizViewModel.questionBankSize * 100).toInt()}%",
+                Toast.LENGTH_LONG
+            )
+        scoreToast.setGravity(Gravity.TOP, 0, 150)
+
+        if (quizViewModel.questionBankAreAnswered) {
+            toastAnswerStatus.show()
+            scoreToast.show()
+        } else {
+            toastAnswerStatus.show()
+        }
```

Comment:
it would be better to extract this to a separate method, i.e `showScoreToast`. You want to avoid methods containing tens of lines of code, better to keep things simple and separated. It`s also easier to test after - imagine if you wanted to test only the code that shows a toast. With the code above, you would have to call `checkAnswer` each time to test the toast, and have potentially unwanted results.

Reply:
done!